src/js/components/DateInput/DateInput.js

```diff
@@ -84,8 +84,13 @@ const DateInput = forwardRef(
     // normalize value based on timestamp vs user's local timezone
     const normalizedDate = normalizeForTimezone(value, timestamp);
 
+    // Check if format is on a range format
+    // 10 characters is the largest non range format (mm/dd/yyyy)
+    // eg. mm-dd-yyyy (non-range) mm/dd/yyyy-mm/dd/yyyy (range)
+    const isFormatRange = format && format.length > 10 && format.includes('-');
```

Comment:
Checking the format length wouldn't flag this format as a range: 'm/d-m/d' since it is less than 10 characters long.
```suggestion
    const isFormatRange = format && format.includes('-') && format.includes('/');
```
I wonder if we should do something like and just not support '-' as a separator when choosing a range of dates since it  might be confusing anyway (we would need to make note of this in the docs). 'm-d-yy-m-d-yy' types of formats are difficult to read and interpret as a range.

Reply:
maybe the range could be a boolean property, and the format is just a regular single format

Reply:
> maybe the range could be a boolean property, and the format is just a regular single format
I think this could cause backwards compatibility issues since we are currently using the format prop to include range formats

Reply:
What about detecting the range condition by looking for duplication of placeholders? For instance, two blocks of 'mm', 'dd', or 'yyyy' would indicate a range scenario.